src/Modulo2.jl

```diff
@@ -271,6 +272,21 @@ function copyto!(a::ZZ2Array, b::ZZ2Array)
     end
 end
 
+"""
+    copyto!(a::ZZ2Vector, num::Integer) 
+
+Copy the bit representation of number `num` into ZZ2Vector `a`
+"""
+function copyto!(a::ZZ2Vector, num::T) where T <: Integer
+    !isbits(num) && throw(ArgumentError("Argument num is not a bits type (BigInt not supported)!"))
+    L = 1 + (sizeof(T) - 1) ÷ sizeof(TA)
+    mask = typemax(TA);
+    @simd for i in 1:L
+        a.data[i] = num & mask
+        num = num >> BA
+    end
```

Comment:
The `@simd` has no effect on my machine. However, `@inbounds` makes a difference.
```suggestion
    for i in 1:L
        @inbounds a.data[i] = num & ~zero(TA)
        num >>= BA
    end
    a
```
Note that I also return the result. We also have to make sure that the padding is remains valid (i.e., padding via zeros only).

Reply:
I thought `@simd` also implies to the complier `@inbounds` as well, but turns out it's the other way around (`@simd` requires all access to be annotated `@inbounds`). Thanks! The padding issue relates to my previous comment on buffer size vs `length(a)`